### PR TITLE
Move get_temporary_prop to platform code

### PR DESCRIPTION
**Pull request by mattmacy**
Signed-off-by: Matt Macy <mmacy@FreeBSD.org>

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: module/os/linux/zfs/zcp_get_os.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+#endif
+
+/*
+ * Takes a dataset, a property, a value and that value's setpoint as
+ * found in the ZAP. Checks if the property has been changed in the vfs.
+ * If so, val and setpoint will be overwritten with updated content.
+ * Otherwise, they are left unchanged.
+ */
+int
+zcp_get_temporary_prop(dsl_dataset_t *ds, zfs_prop_t zfs_prop, uint64_t *val,
```
Comment (behlendorf) [inline module/os/linux/zfs/zcp_get_os.c]: Let's instead move this function after `zfs_register_callbacks()` in `module/os/linux/zfs/zfs_vfsops.c` to co-locate this property handling code.  I'd also suggest naming it `zfs_get_temporary_prop()`.

File: include/sys/zcp_prop.h [hunk 2]
```diff
@@ -26,7 +26,6 @@ extern "C" {
 
 int zcp_load_get_lib(lua_State *state);
 boolean_t prop_valid_for_ds(dsl_dataset_t *ds, zfs_prop_t zfs_prop);
-
```
Comment (behlendorf) [inline include/sys/zcp_prop.h]: nit: no need to change the white space here.

File: module/zfs/zcp_get.c [hunk 3]
```diff
@@ -32,6 +32,7 @@
 #include <sys/sunddi.h>
 #include <sys/zap.h>
 #include <sys/zcp.h>
+#include <sys/zcp_prop.h>
```
Comment (behlendorf) [inline module/zfs/zcp_get.c]: You should no longer need to include this header.

File: module/zfs/zcp_get.c [hunk 4]
```diff
@@ -548,7 +479,7 @@ get_zap_prop(lua_State *state, dsl_dataset_t *ds, zfs_prop_t zfs_prop)
 		    1, &numval, setpoint);
 
 		/* Fill in temporary value for prop, if applicable */
-		(void) get_temporary_prop(ds, zfs_prop, &numval, setpoint);
+		(void) zfs_get_temporary_prop(ds, zfs_prop, &numval, setpoint);
```
Comment (behlendorf) [inline module/zfs/zcp_get.c]: Alternately, how about this to align the code with the quota property handling in `zcp_get_prop()`.    The only potential user space consumer would be ztest and I didn't see a test case which would hit this.

```diff
diff --git a/module/zfs/zcp_get.c b/module/zfs/zcp_get.c
index 0a5f0b824..ccef9d9f4 100644
--- a/module/zfs/zcp_get.c
+++ b/module/zfs/zcp_get.c
@@ -547,8 +547,14 @@ get_zap_prop(lua_State *state, dsl_dataset_t *ds, zfs_prop_t 
                error = dsl_prop_get_ds(ds, prop_name, sizeof (numval),
                    1, &numval, setpoint);
 
-               /* Fill in temorary value for prop, if applicable */
-               (void) get_temporary_prop(ds, zfs_prop, &numval, setpoint);
+#ifdef _KERNEL
+               /* Fill in temporary value for prop, if applicable */
+               (void) zfs_get_temporary_prop(ds, zfs_prop, &numval, setpoint);
+#else
+                return (luaL_error(state,
+                    "temporary properties only supported in kernel mode",
+                    prop_name));
+#endif
 
                /* Push value to lua stack */
                if (prop_type == PROP_TYPE_INDEX) {
```

**Comment by behlendorf** [comment]
@lundman would you mind reviewing this.

**Comment by kithrup** [comment]
LGTM

**Pull request merged**